### PR TITLE
[WIP] Add iPhone ordering example

### PR DESCRIPTION
**Pull request by tsurdilo**
**Many thanks for submitting your Pull Request :heart:!**

**Please specify parts this PR updates:**

- [ x] Specification
- [ ] Schema
- [ x] Examples
- [ ] Extensions
- [ ] Roadmap
- [ ] Use Cases
- [ ] Community
- [ ] TCK
- [ ] Other

**What this PR does / why we need it**:
Adds a new iPhone ordering example.
**Special notes for reviewers**:

**Additional information (if needed):**

### CODE REVIEW

**Comment by tsurdilo** [comment]
depends on https://github.com/serverlessworkflow/specification/pull/271

**Comment by tsurdilo** [comment]
parsed example: https://github.com/tsurdilo/specification/tree/iphoneexample/examples#order-iphone

File: examples/README.md [hunk 1]
```diff
@@ -3483,6 +3484,227 @@ functions:
...
+The purpose of this example is to show how you can implement parallel execution of payment
+processing and preparing the phone shipment. We take advantage of the [Parallel state](../specification.md#Parallel-State)
+and its "completionType" property which is set to "and" by default. This means that all parallel state 
+branches must complete before the state can transition to the next workflow states.
+
+In addition, we show how we can use compensation for each Parallel state branch. This allows us to compensate
+each branch separately. Again, both shipment preparation 
+and payment processing have to complete for the order to be completed successfully. 
+
+For the sake of the example we assume that the "Order Request" event has the following sample payload:
```
Comment (ricardozanini) [inline examples/README.md:3507]:
```suggestion
In this example, we want to implement an iPhone ordering workflow. We receive an order by consuming
an "Order Request" event. Then we start processing the order. 

The purpose of this example is to show how you can implement parallel execution of payment
processing and preparing the phone shipment. We take advantage of the [Parallel state](../specification.md#Parallel-State) and its `completionType` property which is set to `and` by default. This means that all parallel state
branches must complete before the state can transition to the next workflow states.

In addition, we show how we can use compensation for each Parallel state branch. This allows us to compensate
each branch separately. Again, both shipment preparation
and payment processing have to complete for the order to be completed successfully. 

For the sake of the example we assume that the "Order Request" event has the following sample payload:
```

File: examples/README.md [hunk 2]
```diff
@@ -3483,6 +3484,227 @@ functions:
...
+        "phone": "iPhone",
+        "version": "11",
+        "model": "pro",
+        "customerid" : "12345"
+    }
+}
+```
+
+We are not showing the implementation of the function and events definitions in-line. They are assumed 
+to be present in separate json files as shown in the workflow definition.
```
Comment (ricardozanini) [inline examples/README.md:3522]:
```suggestion
We are not showing the implementation of the function and events definitions in-line. They are assumed 
to be present in separate JSON files as shown in the workflow definition.
```

**Review by ricardozanini — APPROVED** [review]

**Comment by tsurdilo** [comment]
@ricardozanini when i do the "apply suggestion" directly DCO fails cause those commits are not signed. do you know if its possible to fix that?

File: examples/README.md [hunk 3]
```diff
@@ -3485,4 +3486,224 @@ functions:
...
+    "states": [
+        {
+            "name": "Receive Order Request",
+            "type": "event",
+            "onEvents": [
+                {
+                    "eventRefs": ["Order Request Event"],
+                    "actions": [
+                        {
+                            "functionRef": "Send order receved confirmation"
```
Comment (jorgenj) [inline examples/README.md:3553]: nit; received

File: examples/README.md [hunk 4]
```diff
@@ -3485,4 +3486,224 @@ functions:
...
+            ],
+            "transition": "Process Order"
+        },
+        {
+            "name": "Process Order",
+            "type": "parallel",
+            "branches": [
+                {
+                    "name": "Prepare Shipment",
+                    "subflow": "Check Shipment Status"
```
Comment (jorgenj) [inline examples/README.md:3566]: This is confusing, I thought this should be the name of another workflow but this example has a state that seems to match this.  Is this supposed to refer to a workflow or a subflow state?

Comment (tsurdilo) [reply examples/README.md:3566]: https://github.com/serverlessworkflow/specification/pull/271

Comment (jorgenj) [reply examples/README.md:3566]: Got it, I found that later and meant to come back and amend my comment but forgot.

**Pull request closed without merging**